Build failure - (64-bit platforms) acrn/bcm958402m2_a72/native_posix_64/... on a number of sanitycheck tests w/TCP2

./scripts/sanitycheck -p acrn

Gets:

```
ERROR   - acrn                      samples/net/sockets/dumb_http_server/sample.net.sockets.dumb_http_server FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/dumb_http_server/sample.net.sockets.dumb_http_server.posix FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/http_get/sample.net.sockets.http_get FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/big_http_download/sample.net.sockets.big_http_download.ci FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/big_http_download/sample.net.sockets.big_http_download FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/http_get/sample.net.sockets.http_get.posix FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/big_http_download/sample.net.sockets.big_http_download.posix FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/websocket_client/sample.net.sockets.websocket_client FAILED: Build failure
ERROR   - acrn                      samples/net/sockets/dumb_http_server_mt/sample.net.sockets.dumb_http_server_mt FAILED: Build failure
ERROR   - acrn                      samples/posix/gettimeofday/sample.posix.gettimeofday FAILED: Build failure
```

```
[ 90%] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp2.c.obj
In file included from /home/galak/git/zephyr/include/logging/log.h:11,
                 from /home/galak/git/zephyr/subsys/net/ip/tcp2.c:7:
/home/galak/git/zephyr/subsys/net/ip/tcp2.c: In function 'net_tcp_put':
/home/galak/git/zephyr/include/logging/log_core.h:238:10: error: format '%u' expects argument of type 'unsigned int', but argument 5 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
  238 |   printk("%c: " fmt "\n", z_log_minimal_level_to_char(_level), \
      |          ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:266:5: note: in expansion of macro 'Z_LOG_TO_PRINTK'
  266 |     Z_LOG_TO_PRINTK(_level, __VA_ARGS__);        \
      |     ^~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:27: note: in expansion of macro 'LOG_DBG'
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                           ^~~~~~~
/home/galak/git/zephyr/subsys/net/ip/tcp2.c:1453:4: note: in expansion of macro 'NET_DBG'
 1453 |    NET_DBG("conn %p pending %u bytes", conn,
      |    ^~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:155:24: error: format '%u' expects argument of type 'unsigned int', but argument 6 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
  155 | #define Z_LOG_STR(...) "%s: " GET_ARG_N(1, __VA_ARGS__), __func__\
      |                        ^~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:279:14: note: in expansion of macro 'Z_LOG_STR'
  279 |              Z_LOG_STR(__VA_ARGS__));\
      |              ^~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:27: note: in expansion of macro 'LOG_DBG'
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                           ^~~~~~~
/home/galak/git/zephyr/subsys/net/ip/tcp2.c:1453:4: note: in expansion of macro 'NET_DBG'
 1453 |    NET_DBG("conn %p pending %u bytes", conn,
      |    ^~~~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:35: error: format '%u' expects argument of type 'unsigned int', but argument 5 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                                   ^~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:188:30: note: in definition of macro '__LOG_INTERNAL'
  188 |    log_from_user(_src_level, __VA_ARGS__);   \
      |                              ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:296:2: note: in expansion of macro '__LOG'
  296 |  __LOG(_level,           \
      |  ^~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:27: note: in expansion of macro 'LOG_DBG'
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                           ^~~~~~~
/home/galak/git/zephyr/subsys/net/ip/tcp2.c:1453:4: note: in expansion of macro 'NET_DBG'
 1453 |    NET_DBG("conn %p pending %u bytes", conn,
      |    ^~~~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:35: error: format '%u' expects argument of type 'unsigned int', but argument 4 has type 'size_t' {aka 'long unsigned int'} [-Werror=format=]
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                                   ^~~~~~~~
/home/galak/git/zephyr/include/logging/log_core.h:291:27: note: in definition of macro '__LOG'
  291 |    log_printf_arg_checker(__VA_ARGS__);         \
      |                           ^~~~~~~~~~~
/home/galak/git/zephyr/include/logging/log.h:71:25: note: in expansion of macro 'Z_LOG'
   71 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/home/galak/git/zephyr/include/net/net_core.h:47:27: note: in expansion of macro 'LOG_DBG'
   47 | #define NET_DBG(fmt, ...) LOG_DBG("(%s): " fmt,    \
      |                           ^~~~~~~
/home/galak/git/zephyr/subsys/net/ip/tcp2.c:1453:4: note: in expansion of macro 'NET_DBG'
 1453 |    NET_DBG("conn %p pending %u bytes", conn,
      |    ^~~~~~~
cc1: all warnings being treated as errors
gmake[2]: *** [zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/build.make:304: zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp2.c.obj] Error 1
```

--- a/subsys/net/ip/tcp2.c
+++ b/subsys/net/ip/tcp2.c
@@ -1450,7 +1450,7 @@ int net_tcp_put(struct net_context *context)
 	if (conn && conn->state == TCP_ESTABLISHED) {
 		/* Send all remaining data if possible. */
 		if (conn->send_data_total > 0) {
-			NET_DBG("conn %p pending %u bytes", conn,
+			NET_DBG("conn %p pending %zu bytes", conn,
 				conn->send_data_total);
 			conn->in_close = true;
 

